cmake/BLTPrivateMacros.cmake

```diff
@@ -842,3 +842,149 @@ macro(blt_clean_target)
+            blt_filter_list(TO _property_list
+                            REGEX "^(INTERFACE_|IMPORTED_LIBNAME_|COMPATIBLE_INTERFACE_|MAP_IMPORTED_CONFIG_)|^(NAME|TYPE|EXPORT_NAME)$"
+                            OPERATION "include")
+        endif()
+
+        ## Print all such properties that have been SET
+        foreach (prop ${_property_list})
+            string(REPLACE "<CONFIG>" "${CMAKE_BUILD_TYPE}" prop ${prop})
+            get_property(_propval TARGET ${arg_TARGET} PROPERTY ${prop} SET)
+            if ("${_propval}")
+                if ((NOT DEFINED arg_PROP_REGEX) OR
+                    (DEFINED arg_PROP_REGEX AND "${prop}" MATCHES "${arg_PROP_REGEX}"))
+                    get_target_property(_propval ${arg_TARGET} ${prop})
+                    message (STATUS "[${arg_TARGET} property] ${prop}: ${_propval}")
+                endif()
+            endif()
+        endforeach()
+        unset(_property_list)
+        unset(_propval)
+    endif()
+
+    ## Additionally, output variables generated via blt_register_target of the form "_BLT_<target>_*"
+    if(_is_blt_registered_target)
+        set(_target_prefix "_BLT_${_target_upper}_")
+
+        ## Filter to get variables of the form _BLT_<target>_ and print
+        get_cmake_property(_variable_names VARIABLES)
+        foreach (prop ${_variable_names})
+            if("${prop}" MATCHES "^${_target_prefix}")
+                if ((NOT DEFINED arg_PROP_REGEX) OR
+                    (DEFINED arg_PROP_REGEX AND "${prop}" MATCHES "${arg_PROP_REGEX}"))
+                    message (STATUS "[${arg_TARGET} property] ${prop}: ${${prop}}")
+                endif()
+            endif()
+        endforeach()
+        unset(_target_prefix)
+        unset(_variable_names)
+    endif()
+
+    unset(_target_upper)
+    unset(_is_blt_registered_target)
+    unset(_is_cmake_target)
+endmacro(blt_print_target_properties_private)
+
+##------------------------------------------------------------------------------
+## blt_find_all_targets_recursive(TARGET <target> TLIST tlist)
+##
+## Store all target's dependancies (link libraries and interface link libraries)
+## recursively in TLIST.
+##
+##------------------------------------------------------------------------------
+macro(blt_find_all_targets_recursive)
+
+    set(options)
+    set(singleValuedArgs TARGET TLIST)
+    set(multiValuedArgs)
+
+    # parse the arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    # check for required arguments
+    if(NOT DEFINED arg_TARGET)
+        message(FATAL_ERROR "TARGET is a required parameter for the blt_find_all_targets_recursive macro")
+    endif()
+
+    if(NOT DEFINED ${arg_TLIST})
+        message(FATAL_ERROR "TLIST is a required parameter for the blt_find_all_targets_recursive macro")
+    endif()
+
+    # check if this is a valid cmake target
+    if(TARGET ${arg_TARGET})
```

Comment:
Add a case here to check if it is a BLT registered library and check the following property for dependencies:

for registered library named `core`, it would be `_BLT_CORE_DEPENDS_ON`

Reply:
This should expose the problem:

```
blt_register_library(NAME bar)
blt_register_library(NAME foo DEPENDS_ON BAR)
blt_print_target_properties(TARGET foo CHILDREN TRUE)
```

Reply:
fixed